theano/tensor/opt.py

```diff
@@ -4189,8 +4189,7 @@ def local_reshape_lift(node):
     """
     if (isinstance(node.op, T.Reshape) and
             node.inputs[0].owner and
-            isinstance(node.inputs[0].owner.op, T.Elemwise) and
-            len(node.inputs[0].owner.inputs) == 1):
+            isinstance(node.inputs[0].owner.op, T.Elemwise)):
         r = node.op(node.inputs[0].owner.inputs[0], node.inputs[1])
```

Comment:
If there are more than 1 input, I think all inputs may have to be reshaped, and if some of them were broadcasted, then the reshape is not going to be as simple as applying `node.op()` on them.